sa/db-next/boulder_sa/20230419000003_OrderToAuthzID.sql

```diff
@@ -0,0 +1,27 @@
+-- +migrate Up
+-- SQL in section 'Up' is executed when this migration is applied
+
+DROP TABLE orderToAuthz2;
+CREATE TABLE `orderToAuthz2` (
+  `id` bigint(20) UNSIGNED NOT NULL AUTO_INCREMENT,
+  `orderID` bigint(20) UNSIGNED NOT NULL,
+  `authzID` bigint(20) UNSIGNED NOT NULL,
+  PRIMARY KEY (`id`),
+  KEY `orderID_idx` (`orderID`),
+  KEY `authzID_idx` (`authzID`)
+) ENGINE=InnoDB AUTO_INCREMENT=9 DEFAULT CHARSET=utf8mb4
```

Comment:
I think the `authzID_idx` is actually not necessary (even though we had it before).

We query the orderToAuthz2 table in two places, both of which should be fine with an index solely on orderID:

https://github.com/letsencrypt/boulder/blob/da8ddc06c1ea7c2c2a160a583c361d4fb25c0a4b/sa/model.go#L1035

https://github.com/letsencrypt/boulder/blob/da8ddc06c1ea7c2c2a160a583c361d4fb25c0a4b/sa/saro.go#L1001

Reply:
Isn't the second of these using the authzID_idx: `ON authz2.ID = orderToAuthxz2.authzID`?

Reply:
I don't think so. I would hope the query planner would use the `orderID` index, given the `orderToAuthz2.orderID = :orderID` clause; and then to match against the authz2 table it would need to use the `ID` index on the `authz2` table; but not the `authzID` index on the `orderToAuthz2` table.

Reply:
I guess the question is: does the query planner prioritize using the "WHERE orderToAuthz2.orderID = foo" clause or using the "ON authz2.ID = orderToAuthz2.orderID" clause to filter rows. My (very vague) impression was that, because this is a left join with the authz2 table on the left, it would do:
1) use WHERE clauses to filter authz2 rows; then
2) use the join to find potential matches in the orderToAuthz2 table; then finally
3) use the rest of the WHERE clauses to filter the joined rows.

But I really have no idea! And I'm not sure how to find out...

Reply:
Here's an `explain` for this query out of my docker database. 
```
MariaDB [boulder_sa_integration]> SET GLOBAL general_log = 'ON';

MariaDB [boulder_sa_integration]> select * from mysql.general_log where argument like '%LEFT JOIN orderToAuthz2 ON%';

MariaDB [boulder_sa_integration]> explain SELECT authz2.id, identifierType, identifierValue, registrationID, status, expires, challenges, attempted, attemptedAt, token, validationError, validationRecord FROM authz2
    -> LEFT JOIN orderToAuthz2 ON authz2.ID = orderToAuthz2.authzID
    -> WHERE authz2.registrationID = 73 AND
    -> authz2.expires > '2023-04-24 19:15:29.474603' AND
    -> authz2.status = 1 AND
    -> orderToAuthz2.orderID = 67;
+------+-------------+---------------+--------+---------------------------------------------------------------------------+---------+---------+----------------------------------------------+------+-------------+
| id   | select_type | table         | type   | possible_keys                                                             | key     | key_len | ref                                          | rows | Extra       |
+------+-------------+---------------+--------+---------------------------------------------------------------------------+---------+---------+----------------------------------------------+------+-------------+
|    1 | SIMPLE      | orderToAuthz2 | ref    | PRIMARY,authzID                                                           | PRIMARY | 8       | const                                        | 1    | Using index |
|    1 | SIMPLE      | authz2        | eq_ref | PRIMARY,regID_expires_idx,regID_identifier_status_expires_idx,expires_idx | PRIMARY | 8       | boulder_sa_integration.orderToAuthz2.authzID | 1    | Using where |
+------+-------------+---------------+--------+---------------------------------------------------------------------------+---------+---------+----------------------------------------------+------+-------------+
2 rows in set (0.001 sec)
```

Reply:
So this shows that it does intend to use the authzID index on the orderToAuthz2 table. In that case we do need to keep it around on this table (above and beyond needing to keep it here to reflect the current reality of prod).

Reply:
I'm fine with this staying or going. If you're curious about getting output from the query planner I believe you can run the query with an `EXPLAIN` statement.

https://mariadb.com/kb/en/explain/

Reply:
Oh lol, jinx @pgporada.